api/src/main/java/org/apache/iceberg/transforms/ProjectionUtil.java

```diff
@@ -52,37 +52,44 @@ private ProjectionUtil() {}
     }
   }
 
-  static UnboundPredicate<Integer> truncateIntegerStrict(
-      String name, BoundPredicate<Integer> pred, Transform<Integer, Integer> transform) {
+  static <T> UnboundPredicate<T> truncateIntegerStrict(
+      String name, BoundPredicate<Integer> pred, Transform<Integer, T> transform) {
     int boundary = pred.literal().value();
     switch (pred.op()) {
       case LT:
-        // predicate would be <= the previous partition
-        return predicate(Expression.Operation.LT_EQ, name, transform.apply(boundary) - 1);
+        // Checking if the literal is at the lower partition boundary
+        if (transform.apply(boundary - 1).equals(transform.apply(boundary))) {
+          // Literal is not at lower boundary, for eg: 123, if width is 10
+          return predicate(Expression.Operation.LT, name, transform.apply(boundary - 1));
```

Comment:
If `transform.apply(boundary - 1)` is equal to `transform.apply(boundary)`, then can't this be replaced with the same as line 66? In that case, I don't think we need the `if` here.

Reply:
🤦‍♀yes, of course. Forgot to change after switching from LT_EQ

Reply:
Done